api/src/sparql.rs

```diff
@@ -0,0 +1,63 @@
+//! Common traits for working with [SPARQL](https://www.w3.org/TR/sparql11-query/).
+//!
+//! # Design rationale
+//!
+//! These traits are deliberately very generic.
+//! Specific implementations may have additional features, such as:
+//! 
+//! - preparing a query for multiple use;
+//! - setting default values for `BASE`, `PREFIX`, `FROM`, `FROM NAMED` directives,
+//!   before parsing query string;
+//! - pre-binding variables before evaluating query;
+//! - etc...
+//!
+//! However, we do not want to impose these feature, or any subset thereof,
+//! to all implementation of Sophia.
+
+use crate::dataset::Dataset;
+use crate::graph::Graph;
+use crate::prefix::PrefixMap;
+use crate::term::TTerm;
+use std::error::Error;
+
+/// A dataset that can be queried with SPARQL.
+pub trait SparqlDataset: Dataset {
+    type Result: SparqlResult;
+    type SparqlError: Error + 'static;
+
+    /// Parse and immediately execute `query`
+    fn execute<'p, P>(&self, query: &str) -> Result<Self::Result, Self::SparqlError>
```

Comment:
I would call this method `query` or `execute_query` to leave room for an implementation of SPARQL update

Reply:
Yes of course, you are right. I'll fix this.